backend/bot/MatchReporter.js

```diff
@@ -1,23 +1,34 @@
-const reportMatchOnSlack = async (bot, match, oldUsers, newUsers) => {
-  const matchResultMessage = createMatchResultMessage(match)
-
-  const oldRankings = oldUsers.sort(ratingComparator)
-  const newRankings = newUsers.sort(ratingComparator)
-
-  const rankingChangeMessage = createRankingChangeMessage(oldRankings, newRankings)
-  await bot.postMessage(`${matchResultMessage}\n${rankingChangeMessage}`)
-  const leaderboardSize = 5
-  const shouldUpdatePurpose = hasLeaderboardChanged(leaderboardSize, oldRankings, newRankings)
-  if (shouldUpdatePurpose) {
-    const purposeMessage = await createPurposeMessage(newRankings.slice(0, leaderboardSize))
-    await bot.setGroupPurpose(purposeMessage)
+const parseMatchResultPrefixSuffixConfig = require('./parseMatchResultPrefixSuffixConfig')
+
+class MatchReporter {
+  constructor(matchResultPrefixSuffixConfig) {
+    try {
+      this.prefixSuffixConfig = parseMatchResultPrefixSuffixConfig(matchResultPrefixSuffixConfig)
+    } catch (e) {
+      console.log(`Parsing matchResultPrefixSuffixConfig failed: ${e.message}`)
+    }
+  }
+
+  async reportMatchOnSlack(bot, match, oldUsers, newUsers) {
+    const matchResultMessage = createMatchResultMessage(match, this.prefixSuffixConfig)
+
+    const oldRankings = oldUsers.sort(ratingComparator)
+    const newRankings = newUsers.sort(ratingComparator)
+
+    const rankingChangeMessage = createRankingChangeMessage(oldRankings, newRankings)
+    await bot.postMessage(`${matchResultMessage}\n${rankingChangeMessage}`)
+    const leaderboardSize = 5
```

Comment:
probably put into construcotr, maybe with a default value of 5...

Reply:
Not my code, but sure.